webhooks/pod/mutate_image_path.go

```diff
@@ -98,60 +107,67 @@ func (ipr *ImagePathRewriter) Handle(ctx context.Context, req admission.Request)
 			if !bound {
 				return admission.Errored(http.StatusInternalServerError, fmt.Errorf("%s of binding %s is empty", annotationProject, psb.Name))
 			}
-
-			rewritePathPrefix := fmt.Sprintf("%s/%s", hsc.Spec.ServerURL, pro)
-			for i, c := range pod.Spec.Containers {
-				ok, err := hasHostDomainPrefix(c.Image)
-				if err != nil {
-					ipr.Log.Error(err, "check host domain", "image", c.Image)
-					continue
-				}
-
-				if !ok {
-					changedC := c.DeepCopy()
-					changedC.Image = rewritePath(rewritePathPrefix, c.Image)
-					pod.Spec.Containers[i] = *changedC
-
-					ipr.Log.Info("rewrite image", "original", c.Image, "rewrite", changedC.Image)
-				} else {
-					ipr.Log.Info("skip container image rewriting as registry host is specified", "image", c.Image)
-				}
-			}
-
-			for i, c := range pod.Spec.InitContainers {
-				ok, err := hasHostDomainPrefix(c.Image)
-				if err != nil {
-					ipr.Log.Error(err, "check host domain", "init_image", c.Image)
-					continue
-				}
-
-				if !ok {
-					changedC := c.DeepCopy()
-					changedC.Image = rewritePath(rewritePathPrefix, c.Image)
-					pod.Spec.InitContainers[i] = *changedC
-
-					ipr.Log.Info("rewrite init image", "original", c.Image, "rewrite", changedC.Image)
-				} else {
-					ipr.Log.Info("skip init container image rewriting as registry host is specified", "image", c.Image)
-				}
+			imageRules := []goharborv1alpha1.ImageRule{
```

Comment:
In this case, we should not only rewrite the image of `BareRegistry`. It seems we need to rewrite all the image paths to the `hsc.pro`.  My follow-up question, how to define rules for every registry?

If the users only want to rewrite the image paths to certain registries, how can we let users defined the rules? It seems the current rule model can not support such cases.

Reply:
You're right, I misunderstood what the original code was doing here.

One possible way we could support a rule for all possible registries would be to use regex instead of matching registry names exactly. Then the rule for the harbor server configuration here would match `*` instead of the `BareRegistry` string. What do you think?